src/pydisagg/ihme/sex_splitter.py

```diff
@@ -0,0 +1,234 @@
+            )
+
+        validate_columns(pattern, self.pattern.columns, name)
+        pattern = pattern[self.pattern.columns].copy()
+        validate_index(pattern, self.pattern.index, name)
+        validate_nonan(pattern, name)
+        validate_positive(pattern, [self.pattern.val_sd], name)
+        data_with_pattern = self._merge_with_pattern(data, pattern)
+        return data_with_pattern
+
+    def get_population_by_sex(self, population, sex_value):
+        return population[population[self.population.sex] == sex_value][
+            self.population.index + [self.population.val]
+        ].copy()
+
+    def parse_population(
+        self, data: DataFrame, population: DataFrame
+    ) -> DataFrame:
+        name = "population"
+        validate_columns(population, self.population.columns, name)
+
+        male_population = self.get_population_by_sex(
+            population, self.population.sex_m
+        )
+        female_population = self.get_population_by_sex(
+            population, self.population.sex_f
+        )
+
+        male_population.rename(
+            columns={self.population.val: "m_pop"}, inplace=True
+        )
+        female_population.rename(
+            columns={self.population.val: "f_pop"}, inplace=True
+        )
+
+        data_with_population = self._merge_with_population(
+            data, male_population, "m_pop"
+        )
+        data_with_population = self._merge_with_population(
+            data_with_population, female_population, "f_pop"
+        )
+
+        validate_nonan(data_with_population, name)
+        return data_with_population
+
+    def _merge_with_population(
+        self, data: DataFrame, population: DataFrame, pop_col: str
+    ) -> DataFrame:
+        data_with_population = data.merge(
+            population,
+            on=self.population.index,
+            how="left",
+        )
+        return data_with_population
+
+    def sex_split_df(
+        self, data: DataFrame, pattern: DataFrame, population: DataFrame
+    ) -> DataFrame:
+        data = self.parse_data(data)
+        data = self.parse_pattern(data, pattern)
+        data = self.parse_population(data, population)
+
+        def sex_split_row(row):
+            split_result, SE = split_datapoint(
+                # This comes from the data
+                observed_total=row[self.data.val],
+                bucket_populations=np.array([row["m_pop"], row["f_pop"]]),
+                # This is from sex_pattern
+                rate_pattern=np.array([1.0, row[self.pattern.val]]),
+                model=RateMultiplicativeModel(),
+                output_type="rate",
+                normalize_pop_for_average_type_obs=True,
+                # This is from the data
+                observed_total_se=row[self.data.val_sd],
+                # This is from sex_pattern
+                pattern_covariance=np.diag(
+                    np.array([0, row[self.pattern.val_sd] ** 2])
+                ),
+            )
```

Comment:
This is ok for now, later we'll want to add a tag to allow for logit differences. Created issue #59 for this.